src/Shared/ServerInfrastructure/StringUtilities.cs

```diff
@@ -17,102 +18,211 @@ internal class StringUtilities
-                    else
+                }
+
+                if (input <= end - Vector128<sbyte>.Count)
+                {
+                    Vector128<sbyte> zero = Vector128<sbyte>.Zero;
+
+                    do
                     {
-                        // 32-bit: Loop ints by default
-                        while (input <= end - sizeof(int))
+                        var vector = Sse2.LoadVector128(input).AsSByte();
+                        if (!CheckBytesInAsciiRange(vector, zero))
                         {
-                            isValid &= CheckBytesInAsciiRange(((int*)input)[0]);
+                            return false;
+                        }
 
-                            output[0] = (char)input[0];
-                            output[1] = (char)input[1];
-                            output[2] = (char)input[2];
-                            output[3] = (char)input[3];
+                        var c0 = Sse2.UnpackLow(vector, zero).AsUInt16();
+                        var c1 = Sse2.UnpackHigh(vector, zero).AsUInt16();
 
-                            input += sizeof(int);
-                            output += sizeof(int);
-                        }
+                        Sse2.Store((ushort*)output, c0);
+                        Sse2.Store((ushort*)output + Vector128<ushort>.Count, c1);
+
+                        input += Vector128<sbyte>.Count;
+                        output += Vector128<sbyte>.Count;
+                    } while (input <= end - Vector128<sbyte>.Count);
+
+                    if (input == end)
+                    {
+                        return true;
+                    }
+                }
+            }
+            else if (Vector.IsHardwareAccelerated)
+            {
+                while (input <= end - Vector<sbyte>.Count)
+                {
+                    var vector = Unsafe.AsRef<Vector<sbyte>>(input);
+                    if (!CheckBytesInAsciiRange(vector))
+                    {
+                        return false;
+                    }
+
+                    Vector.Widen(
+                        vector,
+                        out Unsafe.AsRef<Vector<short>>(output),
+                        out Unsafe.AsRef<Vector<short>>(output + Vector<short>.Count));
+
+                    input += Vector<sbyte>.Count;
+                    output += Vector<sbyte>.Count;
+                }
+
+                if (input == end)
+                {
+                    return true;
+                }
+            }
+
+            if (Environment.Is64BitProcess) // Use Intrinsic switch for branch elimination
+            {
+                // 64-bit: Loop longs by default
+                while (input <= end - sizeof(long))
+                {
+                    var value = *(long*)input;
+                    if (!CheckBytesInAsciiRange(value))
+                    {
+                        return false;
+                    }
+
+                    if (Bmi2.X64.IsSupported)
```

Comment:
@GrabYourPitchforks with respect to https://github.com/dotnet/runtime/issues/2251 and your PR https://github.com/dotnet/runtime/pull/31904 should `Bmi2` be removed here too?
I don't know what's the current stance on this.

Reply:
Yes, BMI2 should be removed if possible. For AMD processors this PR is likely to introduce similar regressions as those we are fixing via https://github.com/dotnet/runtime/issues/2251.

Reply:
-> https://github.com/dotnet/aspnetcore/issues/18949 to track this.